x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -193,10 +199,11 @@ public void getModelForSearch(String modelId, ActionListener<Model> modelActionL
      * @param consumer            which feature is requesting the model
      * @param modelActionListener the listener to alert when the model has been retrieved.
      */
-    private void getModel(String modelId, Consumer consumer, ActionListener<Model> modelActionListener) {
+    private void getModel(String modelId, Consumer consumer, ActionListener<LocalModel> modelActionListener) {
         ModelAndConsumer cachedModel = localModelCache.get(modelId);
         if (cachedModel != null) {
             cachedModel.consumers.add(consumer);
+            cachedModel.model.acquire();
             modelActionListener.onResponse(cachedModel.model);
             logger.trace(() -> new ParameterizedMessage("[{}] loaded from cache", modelId));
             return;
```

Comment:
This might have a race condition (will think more)

- Model is grabbed from cache in 203. Model has a ref count of 1
- Model is evicted from cache due to size or something
- Ref count is now 0
- Model is attempted to acquire again but ref count 0 and the bytes are already removed from the breaker

Reply:
I made a change so that if the ref count == 0 then the model bytes are added back to the circuit breaker. In some cases this means `acquire` could throw so I have to handle that